PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextEnabledHandler.cs

```diff
@@ -10,7 +13,22 @@ class HighlightTextEnabledHandler : EnabledHandler
     {
         protected override bool GetEnabled(string ribbonId)
         {
-            return this.GetRibbonUi().HighlightTextFragmentsEnabled;
+            try
+            {
+                if (this.GetAddIn().Application.ActiveWindow.Selection.Type == PpSelectionType.ppSelectionText && 
+                    this.GetAddIn().Application.ActiveWindow.Selection.TextRange2.TrimText().Length > 0)
+                {
+                    return this.GetRibbonUi().HighlightTextFragmentsEnabled;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (System.Runtime.InteropServices.COMException) // If this exception is caught, it means nothing has been selected yet
```

Comment:
shift the comments to a new line, see Commenting Conventions [here](https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-CSharp.adoc)

 
--